modules/dev-server/main.tf

```diff
@@ -173,7 +243,51 @@ exec consul agent -server \
   -hcl 'telemetry { disable_compat_1.9 = true }' \
   -hcl 'connect { enabled = true }' \
   -hcl 'enable_central_service_config = true' \
+%{if var.tls~}
+  -hcl='ca_file = "/consul/consul-agent-ca.pem"' \
+  -hcl='cert_file = "/consul/dc1-server-consul-0.pem"' \
+  -hcl='key_file = "/consul/dc1-server-consul-0-key.pem"' \
+  -hcl='auto_encrypt = {allow_tls = true}' \
+  -hcl='ports { https = 8501 }' \
+  -hcl='verify_incoming_rpc = true' \
+  -hcl='verify_outgoing = true' \
+  -hcl='verify_server_hostname = true' \
+%{endif~}
 EOF
+
+  consul_server_tls_init_command = <<EOF
+ECS_IPV4=$(curl -s $ECS_CONTAINER_METADATA_URI | jq -r '.Networks[0].IPv4Addresses[0]')
+cd /consul
+echo "$CONSUL_CACERT" > ./consul-agent-ca.pem
+echo "$CONSUL_CAKEY" > ./consul-agent-ca-key.pem
+consul tls cert create -server -additional-ipaddress=$ECS_IPV4
```

Comment:
Worth a comment saying we need this init container to generate the server cert because the IP of the task must be included in the SAN and we don't know the IP ahead of time.